AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/InitialSync/InitialSyncOperation.swift

```diff
@@ -141,23 +152,23 @@ final class InitialSyncOperation: AsynchronousOperation {
         }
 
         let items = syncQueryResult.items
-        lastSyncTime = syncQueryResult.startedAt
+        recordsReceived += UInt(items.count)
 
         for item in items {
             reconciliationQueue.offer(item)
         }
 
-        if let nextToken = syncQueryResult.nextToken {
+        if let nextToken = syncQueryResult.nextToken, recordsReceived < syncMaxRecords {
             DispatchQueue.global().async {
                 self.query(nextToken: nextToken)
             }
         } else {
-            updateModelSyncMetadata()
+            updateModelSyncMetadata(lastSyncTime: syncQueryResult.startedAt)
         }
 
     }
 
-    private func updateModelSyncMetadata() {
+    private func updateModelSyncMetadata(lastSyncTime: Int?) {
```

Comment:
I'm not seeing that you are using `lastSyncTime` in this method?

Reply:
On line 182.  But hmm.. seems like it's not a good idea to overlap the local function variable `lastSyncTime` with the instance variable of `lastSyncTime`.

We were previously always setting the `lastSyncTime`, which is a bug, because we can have a case where we need to paginate, and when we paginate through a base query, we do not want `lastSyncTime` to be populated.

But, this has made me think....to make things more clear, it seems like we don't need to keep `lastSyncTime` as an instance variable.

Reply:
Updated to remove instance variable, and will hopefully make things clearer! thanks for the input!